docs/design/2019-11-14-tidb-builtin-diagnostics-zh_CN.md

```diff
@@ -0,0 +1,680 @@
+
+#### 可复用的 HTTP API
+
+目前 TiDB/TiKV/PD 包含部分可复用 HTTP API，本提案暂不将对应接口迁移至 gRPC Service，迁移工作由后续其他计划完成。所有 HTTP API 需要以 JSON 格式返回数据，以下是提案中可能用到的 HTTP API 列表：
+
+- 获取配置信息
+    - PD: /pd/api/v1/config
+    - TiDB/TiKV: /config
+- 性能采样接口: TiDB/PD 包含以下所有接口，TiKV 暂时只包含 CPU 性能采样接口
+    - CPU: /debug/pprof/profile
+    - Memory: /debug/pprof/heap
+    - Allocs: /debug/pprof/allocs
+    - Mutex: /debug/pprof/mutex
+    - Block: /debug/pprof/block
+
+### 集群信息系统表
+
+每个 TiDB 实例均可以通过前两层提供的 HTTP API 或 gRPC Service 访问其他节点的信息，从而实现集群的 Global View。本提案中通过新建一系列相关系统表将采集到的集群信息向上层提供数据，上层包括不限于：
+
+- 终端用户：用户直接通过 SQL 查询获取集群信息排查问题
+- 运维系统：TiDB 的使用环境比较多样，客户可以通过 SQL 获取集群信息将 TiDB 集成到自己的运维系统中
+- 生态工具：外部工具通过 SQL 拿到集群信息实现功能定制，比如 `sqltop` 可以直接通过集群 `events_statements_summary_by_digest` 获取整个集群的 SQL 采样信息
+
+#### 集群拓扑系统表
+
+要为 TiDB 实例提供一个 **Global View**，首先需要为 TiDB 实例提供一个拓扑系统表，可以从拓扑系统表中获取各个节点的 HTTP API Address 和 gRPC Service Address，从而方便的构造出各个远程 API 的 Endpoint，进一步获取目标节点采集的信息。
+
+本提案实现完成可以通过 SQL 查询以下结果：
+
+```
+mysql> use information_schema;
+Database changed
+
+mysql> desc TIDB_CLUSTER_INFO;
++----------------+---------------------+------+------+---------+-------+
+| Field          | Type                | Null | Key  | Default | Extra |
++----------------+---------------------+------+------+---------+-------+
+| ID             | bigint(21) unsigned | YES  |      | NULL    |       |
+| TYPE           | varchar(64)         | YES  |      | NULL    |       |
+| NAME           | varchar(64)         | YES  |      | NULL    |       |
+| ADDRESS        | varchar(64)         | YES  |      | NULL    |       |
+| STATUS_ADDRESS | varchar(64)         | YES  |      | NULL    |       |
+| VERSION        | varchar(64)         | YES  |      | NULL    |       |
+| GIT_HASH       | varchar(64)         | YES  |      | NULL    |       |
++----------------+---------------------+------+------+---------+-------+
+7 rows in set (0.00 sec)
+
+mysql> select TYPE, ADDRESS, STATUS_ADDRESS,VERSION from TIDB_CLUSTER_INFO;
++------+-----------------+-----------------+-----------------------------------------------+
+| TYPE | ADDRESS         | STATUS_ADDRESS  | VERSION                                       |
++------+-----------------+-----------------+-----------------------------------------------+
+| tidb | 127.0.0.1:4000  | 127.0.0.1:10080 | 5.7.25-TiDB-v4.0.0-alpha-793-g79eef48a3-dirty |
+| pd   | 127.0.0.1:2379  | 127.0.0.1:2379  | 4.0.0-alpha                                   |
+| tikv | 127.0.0.1:20160 | 127.0.0.1:20180 | 4.0.0-alpha                                   |
++------+-----------------+-----------------+-----------------------------------------------+
+3 rows in set (0.00 sec)
+```
+
+#### 监控信息系统表
+
+由于监控指标会随着程序的迭代添加和删除监控指标，对于同一个监控指标，可能有不同的表达式获取监控不同维度的信息。鉴于以上两个需求，需要设计一个有弹性的监控系统表框架，本提案暂时才采取以下方案：将表达式映射为 `metrics_schema` 数据库中的系统表，表达式与系统表的关系可以通过以下方式关联：
+
+- 定义在配置文件
+
+    ```
+    # tidb.toml
+    [metrics_schema]
+    qps = `sum(rate(tidb_server_query_total[$INTERVAL] offset $OFFSET_TIME)) by (result)`
```

Comment:
em, seem we want to deprecate the config file soon, so maybe we can add some basic metric schemas in Go code directly, then we can use HTTP API or SQL load to add more or update exiting ones.

Reply:
Do we need to delete this section? or keep it in the proposal but not implement it.